test/performance/operf.go

```diff
@@ -0,0 +1,634 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type metric struct {
+	Max float64
+	Avg float64
+}
+
+type lvmUnitMetrics struct {
+	Namespace    string
+	Name         string
+	WorkloadType string
+	cpu          metric
+	ram          metric
+	fs           metric
+	start        time.Time
+	end          time.Time
+}
+
+const (
+	cpuQueryTemplate   = `sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate{cluster="", namespace="{{.Namespace}}"} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	memQueryTemplate   = `sum(container_memory_working_set_bytes{cluster="", namespace="{{.Namespace}}",container!="", image!=""} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	filesystemTemplate = `sum(pod:container_fs_usage_bytes:sum * on(pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	Mebibyte           = 1048576 // bytes
+)
+
+const (
+	usageHelp = `Usage of operf:
+This script provide LVM operator metrics retrieval (for all the units in the operator) doing two different kind of tests:
+
+- LVMcluster creation:
+	This implies deployment of daemonsets in the nodes selected and the LVM Volume Groups creation.
+	Example:
+	# go run test/performance/operf.go -token sha256~cj81ClyUYu7g05y8K-uLWm2AbrKTbNEQ96hEJcWStQo -lvmcrd config/samples/lvm_v1alpha1_lvmcluster.yaml
+
+- PVCs and PVs creation and usage:
+	It is created <number> of "busy" pods using PVCs bounded to PVs which use the Storage class provided
+	Example:
+	# go run test/performance/operf.go -token sha256~cj81ClyUYu7g05y8K-uLWm2AbrKTbNEQ96hEJcWStQo -number 16
+
+Parameters:`
+)
+
+var (
+	// flags
+	testNamespace        *string
+	testStorageClassName *string
+	namePattern          *string
+	numberInstances      *int
+	lvmClusterFile       *string
+	token                *string
+
+	// variables
+	PVCLabels     = map[string]string{"app": "testPerformance"}
+	prometheusURL string
+)
+
+func main() {
+	testNamespace = flag.String("namespace", "lvm-operator-system", "Namespace where the PVCs and test pods will be deployed/undeployed")
```

Comment:
Looks like the same namespace where operator is deployed.  The help text should reflect this.

Reply:
done